lib/global-config.js

```diff
@@ -115,25 +122,108 @@ if (exports.config.github) {
   dprepend(exports.config.registries.github, exports.config.github);
   delete exports.config.github;
 }
-save();
 
-exports.set = function(name, val) {
-  var nameParts = name.split('.');
 
-  var config = exports.config;
-  var part;
-  while (nameParts.length > 1) {
-    part = nameParts.shift();
-    config[part] = typeof config[part] === 'object' ? config[part] : {};
-    config = config[part];
+save(exports.config, isLocal());
+
+
+
+function loadConfigFile(configFilePath) {
+    var config;
+    try {
+      config = JSON.parse(fs.readFileSync(configFilePath).toString() || '{}');
+    }
+    catch(e) {
+      config = {};
+      ui.log('err', 'Unable to read ' + config.current + ' configuration file.\nEnsure you have read and write access to %' + configFilePath + '%.');
+      process.exit(1);
+    }
+
+    return config;
+}
+
+
+function getLocalConfigPath() {
+    var configDir,
+        configFilePath;
+
+    if ( process.env && typeof process.env.jspmConfigPath === 'string' ) {
+        configDir = process.env.jspmConfigPath.substr(0, process.env.jspmConfigPath.lastIndexOf('/'));
+    } else {
+        configDir =  process.cwd();
+    }
+
+    if ( fs.existsSync(configDir + path.sep + '.jspmrc') ) {
+        configFilePath = configDir + path.sep + '.jspmrc';
+    } else if ( fs.existsSync(configDir + path.sep + '.config' + path.sep + '.jspmrc') ) {
+        configFilePath = configDir + path.sep + '.config' + path.sep + '.jspmrc';
+    }
+
+    return configFilePath;
+}
+
+function getDefaultLocalConfigPath() {
+    var configDir = process.env.jspmConfigPath.substr(0, process.env.jspmConfigPath.lastIndexOf('/')) || process.cwd();
+
+    return configDir + path.sep + '.jspmrc';
+}
+
+
+function isLocal() {
+    return config.current === 'local';
+}
+
+
+exports.save = save;
+function save(configToWrite, saveLocal) {
+  var place = saveLocal === true ? 'local' : 'global';
+
+  if ( place === 'global' ) {
+    try {
+      fs.mkdirSync(HOME + path.sep + '.jspm');
+    }
+    catch(e) {
+    if (e.code !== 'EEXIST')
+      throw 'Unable to create jspm system folder\n' + e.stack;
+    }
+  }
+
+  // If the user wants to save options to a local configuration file that is not
+  // created, we need to create it and extend it with the default config.
+  if ( saveLocal && !config.local.path ) {
```

Comment:
Ok, found the reason why I added it... Without that if clause, if I try to save local config in a dir where no local config file exists, [L199](https://github.com/hmps/jspm-cli/blob/local-config/lib/global-config.js#L199) will throw because `config.local.path` is undefined.

The reason is that `config.local.path`, set on [L54](https://github.com/hmps/jspm-cli/blob/local-config/lib/global-config.js#L54) via `getLocalConfigPath()`, will be not be set if no local config file exists. So in the `if`, I `getDefaultLocalConfigPath()` to know where to save the generated config file.

That `config.local.path` is undefined is used to know which config to load ([L87](https://github.com/hmps/jspm-cli/blob/local-config/lib/global-config.js#L87)), that's why I don't set it to a default value in `getLocalConfigPath`.

Does that make sense to you @guybedford?